a4_APIweatherApp.py

```diff
@@ -0,0 +1,90 @@
+'''
+This program is inteded to use the Open Meteo API
+and output an API and Mock plot of temperatures in the next 24 hours
+'''
+
+from abc import ABC, abstractmethod
+import requests
+import matplotlib.pyplot as plt
+
+class Service(ABC):
+    @abstractmethod
+    def getting(self):
+        pass
+
+class mockService(Service):
+    def getting(self):
+        #create a 
+        dates = ["2024-08-02T02:00",
+      "2024-08-02T03:00",
+      "2024-08-02T04:00",
+      "2024-08-02T05:00",
+      "2024-08-02T06:00",
+      "2024-08-02T07:00",
+      "2024-08-02T08:00",
+      "2024-08-02T09:00",
+      "2024-08-02T10:00",
+      "2024-08-02T11:00",
+      "2024-08-02T12:00",
+      "2024-08-02T13:00",
+      "2024-08-02T14:00",
+      "2024-08-02T15:00",
+      "2024-08-02T16:00",
+      "2024-08-02T17:00",
+      "2024-08-02T18:00",
+      "2024-08-02T19:00",
+      "2024-08-02T20:00",
+      "2024-08-02T21:00",
+      "2024-08-02T22:00",
+      "2024-08-02T23:00",
+      "2024-08-03T00:00",
+      "2024-08-03T01:00"]
+        temps = [19.4, 18.8, 18.2, 17.8, 17.7, 17.7, 17.9, 18, 18.1, 18.6, 18.8, 18.8, 18.9, 19.2, 19.4, 19.6, 19.4, 19.5, 19.4, 19, 18.5, 18, 17.6, 17.3]
+        plt.scatter(dates, temps)
+        plt.title("Mock Service")
+        plt.show()
+
+
+class APIService(Service):
+    def getting(self):
+        response = requests.get("https://api.open-meteo.com/v1/forecast?latitude=52.52&longitude=13.41&hourly=temperature_2m")
```

Comment:
Here, you've hardcoded the parameters to the API query as well, which is not great since this mean you are always looking up weather data for a single lat/lon. This function should definitely allow the client of this function to pass in the lat/lon values. For the main function of the script, you could choose to keep it simple and just keep on value hardcoded there, but doing it here does not make sense.

Keeping a constant URL is fine, but the parameters for it should be passed in through variables. For example, you can pass non-hardcoded params to any HTTP request like this

```
        base_url = "https://api.open-meteo.com/v1/forecast"
        params = {}
        params['latitude'] = latitude    # float
        params['longitude'] = longitude    # float
        params['hourly'] = daily_weather_variable    # string with "temperature_2m"

        response = requests.get(base_url, params=params)
```